coalib/results/result_actions/GeneratePatchesAction.py

```diff
@@ -0,0 +1,89 @@
+from pygments.lexers import guess_lexer_for_filename
+from pyprint.ConsolePrinter import ConsolePrinter
+
+from coalib.results.result_actions.ResultAction import ResultAction
+from coalib.results.result_actions.ShowPatchAction import ShowPatchAction
+from coalib.output.printers.LogPrinter import LogPrinter
+from coalib.parsing.FilterHelper import FilterHelper
+from coalib.coala_modes import mode_normal
+from coalib.parsing.DefaultArgParser import default_arg_parser
+
+
+OBJECT_INDEX = FILENAME_INDEX = 0
+DEFAULTBEAR = 'SpaceConsistencyBear'
+
+
+def show_possibilities(console_printer, i, action):
+    console_printer.print('[{:>4}]  {}. Apply patch (\'{}\')'.format('', i,
+                                                                     action))
+
+
+def create_arg_parser(files, bears):
+    """
+    A function that generates a `default_arg_parser`
+
+    :param files: A list that contains filenames.
+    :param bears: A list that contains name of bears.
+    :return:      An object of type `default_arg_parser`.
+    """
+    args = default_arg_parser().parse_args()
+    args.files = files
+    args.bears = bears
+    args.default_actions = '*: ApplyPatchAction'
+
+    return args
+
+
+class DefaultBear():
+    def __init__(self):
+        self.name = DEFAULTBEAR
+
+
+class GeneratePatchesAction(ResultAction):
+    SUCCESS_MESSAGE = 'Patch generated successfully.'
+
+    is_applicable = staticmethod(ShowPatchAction.is_applicable)
+
+    def apply(self,
+              result,
+              original_file_dict,
+              file_diff_dict):
+        """
+        (G)enerate patches
+        """
+
+        console_printer = ConsolePrinter()
+        possible_options = []
+        log_printer = LogPrinter()
+        to_filename = sorted(result.diffs.items())[OBJECT_INDEX][FILENAME_INDEX]
+
+        # Find the language used in the file provided by the user
+        language = guess_lexer_for_filename(to_filename, 'Error, no file '
+                                                         'found').name
+
+        # Find all the bears that can run with the language found
+        filtered_bears = FilterHelper.apply_filters(
+            [['language', language]], None)[0]['cli']
+
+        filtered_bears = list(filtered_bears)
+        filtered_bears.insert(0, DefaultBear())
+        for b in filtered_bears:
+            possible_options.append(b.name)
```

Comment:
`possible_options = [b.name for b in filtered_bears]`
You can also remove first definition of `possible_options = []` as its not being used in between.